Fixing broken links in MTA docs

RichardHoch opened the pull request:
Fixes the broken links in MTA docs issue raised in https://issues.redhat.com/browse/WINDUP-3014. 

Previews:

For the repaired link to _Introduction to the Migration Tools for Applications_ guide, formerly the _Getting Started Guide_:
http://file.emea.redhat.com/rhoch/01032021/clig/html-single/#about_mta_cli-guide
http://file.emea.redhat.com/rhoch/01032021/maveng/html-single/#about_mta_maven-guide
http://file.emea.redhat.com/rhoch/01032021/pluging/html-single/#about_mta_plugin-guide
http://file.emea.redhat.com/rhoch/01032021/rulesg/html-single/#about_rules_dev_guide_rules-development-guide
http://file.emea.redhat.com/rhoch/01032021/webcg/html-single/#about_mta_web-console-guide

For the repaired link to the IDE plugin download:
http://file.emea.redhat.com/rhoch/01032021/pluging/html-single/#installing-web-console_plugin-guide-offline

Installation procedures for the web console, CLI, and IDE plugin:

http://file.emea.redhat.com/rhoch/01032021/webcg/html-single/#installing_the_web_console
http://file.emea.redhat.com/rhoch/01032021/pluging/html-single/#installing-web-console_plugin-guide
http://file.emea.redhat.com/rhoch/01032021/clig/html-single/#installing-web-console_cli-guide



_Intoduction to the Migration Toolkit for Applications_  http://file.emea.redhat.com/rhoch/01032021/gsg/html-single/

RichardHoch commented:
@apinnick: please review.

apinnick commented on docs/topics/templates/document-attributes.adoc:
```diff
@@ -83,7 +83,7 @@
 
 // URL for downloads on developers.redhat.com
-:ProductDownloadURL: https://developers.redhat.com/products/
+:ProductDownloadURL: https://developers.redhat.com/download-manager/file/
```
Direct links are way too much work to maintain and very tricky. Sometimes a build does not get uploaded to the dev site in time, which means the link is temporarily broken.

I recommend that you change all the direct download links to this:

"Navigate to the `Migration Toolkit for Applications download` site[https://developers.redhat.com/products/mta/download] and download the ...."

At least that way, if the new build has not been uploaded, the user can bookmark the page or choose to download an older version of the file.

RichardHoch commented:
@apinnick -- something is odd with the link in line 148 of cli-args.adoc.  I changed the link when resolving https://issues.redhat.com/browse/WINDUP-3033, a Jira that come in yesterday as I was working on this PR. I saw that in terms of the document referenced, Jira 3033 relates to the same broken document link as the _Getting Started_ /_Introduction to the Migration_Toolkit_for_Applications_ issue in this PR -- the broken link led to a 404 message, which i fixed.

I noticed that once I resolved the 404 issue, the link did not point to the exact section desired of the document: https://access.redhat.com/documentation/en-us/migration_toolkit_for_applications/5.1/html-single/introduction_to_the_migration_toolkit_for_applications/supported_configurations#migration_paths_getting-started-guide. Instead, the link only led to the document, not the correct section, meaning it needed additional work.

So I changed the link in line 148 of cli-args.adoc accordingly. When I previewed the rendered document using master.adoc, the link (in the warning message) worked perfectly, but in the bccutil and preview versions of the CLI guide (see the link in the warning message in http://file.emea.redhat.com/rhoch/01032021/clig/html-single/#target_argument_cli-guide), the document opens but not the correct section. (https://access.redhat.com/documentation/en-us/migration_toolkit_for_applications/5.1/html-single/introduction_to_the_migration_toolkit_for_applications/index#migration_paths). [Ctrl+Shift+A in any master.adoc lets you see a fully rendered version of the whole document, unlike when use the command with other .adoc files].

I have spent at least 2 hours on making small changes to the link in cli-args.adoc, but I still cannot get it to point correctly to the right section (https://access.redhat.com/documentation/en-us/migration_toolkit_for_applications/5.1/html-single/introduction_to_the_migration_toolkit_for_applications/index#migration_paths_getting-started-guide). Could you please take a look when you have time?

apinnick commented:
> @apinnick -- something is odd with the link in line 148 of cli-args.adoc. I changed the link when resolving https://issues.redhat.com/browse/WINDUP-3033, a Jira that come in yesterday as I was working on this PR. I saw that in terms of the document referenced, Jira 3033 relates to the same broken document link as the _Getting Started_ /_Introduction to the Migration_Toolkit_for_Applications_ issue in this PR -- the broken link led to a 404 message, which i fixed.
> 
> I noticed that once I resolved the 404 issue, the link did not point to the exact section desired of the document: https://access.redhat.com/documentation/en-us/migration_toolkit_for_applications/5.1/html-single/introduction_to_the_migration_toolkit_for_applications/supported_configurations#migration_paths_getting-started-guide. Instead, the link only led to the document, not the correct section, meaning it needed additional work.
> 
> So I changed the link in line 148 of cli-args.adoc accordingly. When I previewed the rendered document using master.adoc, the link (in the warning message) worked perfectly, but in the bccutil and preview versions of the CLI guide (see the link in the warning message in http://file.emea.redhat.com/rhoch/01032021/clig/html-single/#target_argument_cli-guide), the document opens but not the correct section. (https://access.redhat.com/documentation/en-us/migration_toolkit_for_applications/5.1/html-single/introduction_to_the_migration_toolkit_for_applications/index#migration_paths). [Ctrl+Shift+A in any master.adoc lets you see a fully rendered version of the whole document, unlike when use the command with other .adoc files].
> 
> I have spent at least 2 hours on making small changes to the link in cli-args.adoc, but I still cannot get it to point correctly to the right section (https://access.redhat.com/documentation/en-us/migration_toolkit_for_applications/5.1/html-single/introduction_to_the_migration_toolkit_for_applications/index#migration_paths_getting-started-guide). Could you please take a look when you have time?

Resolved

RichardHoch commented:
@apinnick: please review. I ran the link checker again to be sure all links are OK.

apinnick commented on docs/topics/installing-web-console.adoc:
```diff
@@ -120,7 +120,7 @@ $ mvn clean install
 endif::[]
 ifdef::plugin-guide-offline[]
-. Download the link:{ProductDownloadURL}{IDEPluginFilename}-{ProductVersion}.zip[IDE Plugin archive file].
+. Navigate to  the link:{DownloadPageURL}[*Migration Toolkit for Applications* site] and download the appropriate IDE plugin.
```
Remove extra space.
Also, it should be styled "Migration Toolkit for Applications download site", no bold.

apinnick commented on docs/topics/templates/document-attributes.adoc:
```diff
@@ -83,7 +83,7 @@
 // URL for downloads on developers.redhat.com
-:ProductDownloadURL: https://developers.redhat.com/products/
+:DownloadPageURL: https://developers.redhat.com/products/mta/download
 
```
Did you remove all download URLs that pointed to specific products on the developers.redhat.com site?

RichardHoch replied:
@apinnick: Created the following attributes, to cover all current download procedures:

:MTADownloadPageURL: https://developers.redhat.com/products/mta/download
:CodeReadyStudioDownloadPageURL: https://developers.redhat.com/products/codeready-studio/download/

Updated the relevant URLs as well as the procedure texts: 

http://file.emea.redhat.com/rhoch/01032021/webcg/html-single/#installing_the_web_console
http://file.emea.redhat.com/rhoch/01032021/pluging/html-single/#installing-web-console_plugin-guide

Please review.

apinnick commented on docs/topics/installing-web-console-on-openshift.adoc:
```diff
@@ -41,7 +41,7 @@ endif::[]
 .Procedure
 
-. Download the installation archive file from the link:https://developers.redhat.com/products/mta/download[{ProductShortName} download page].
+. Navigate to the  link:{MTADownloadPageURL}[{ProductShortName} Download page] and download the installation archive file.
```
Be a bit more specific than "installation archive file". 

"{ProductShortName} installation file"

apinnick commented on docs/topics/installing-web-console.adoc:
```diff
@@ -120,7 +120,7 @@ $ mvn clean install
 endif::[]
 ifdef::plugin-guide-offline[]
-. Download the link:{ProductDownloadURL}{IDEPluginFilename}-{ProductVersion}.zip[IDE Plugin archive file].
+. Navigate to the link:{MTADownloadPageURL}[Migration Toolkit for Applications site] and download the appropriate IDE plugin.
```
[Migration Toolkit for Applications site] > [Migration Toolkit for Applications download site]

appropriate IDE plugin > IDE Plugin Repository file

apinnick commented on docs/topics/installing-web-console.adoc:
```diff
@@ -63,7 +63,7 @@ endif::[]
 
 ifdef::cli-guide,web-console-guide[]
-. Download the installation archive file from the link:https://developers.redhat.com/products/mta/download[{ProductShortName} Download page].
+. Navigate to the  link:{MTADownloadPageURL}[{ProductShortName} Download page] and download the installation archive file.
```
installation archive file > {ProductShortName} installation file

apinnick commented on docs/topics/rules-guide-intro.adoc:
```diff
@@ -5,6 +5,6 @@
 This guide is for engineers, consultants, and others who want to create custom XML-based rules for {ProductName} ({ProductShortName}) tools.
 
-If you are new to {ProductShortName}, it is recommended that you start with the link:{ProductDocUserGuideURL}[_{GettingStartedBookName}_] for an overview of the {ProductName} features and system requirements. It is also recommended that you review the link:{ProductDocUserGuideURL}[_{UserCLIBookName}_], which provides detailed instructions on how to install and execute the {CLIName}.
+If you are new to {ProductShortName}, it is recommended that you start with the link:{ProductDocGettingStartedGuideURL}[_{GettingStartedBookName}_] for an overview of the {ProductName} features and system requirements. It is also recommended that you review the link:{ProductDocUserGuideURL}[_{UserCLIBookName}_], which provides detailed instructions on how to install and execute the {CLIName}.
```
We don't say "It is recommended". Rewrite so that this does not appear. IBM

While you're at it, you should update the attribute name so that future TWs don't make the same mistake as you did -- not realizing that "Getting Started" = Introduction to.

apinnick commented on docs/topics/about-rules.adoc:
```diff
@@ -19,5 +19,5 @@ otherwise(action)
 {ProductShortName} provides a comprehensive set of standard migration rules out-of-the-box. Because applications may contain custom libraries or components, {ProductShortName} allows you to write your own rules to identify use of components or software that may not be covered by the existing ruleset.
 
 ifndef::rules-development-guide[]
```
Why does this ifndef occur if the module is used ONLY in the RD guide?

RichardHoch replied:
The module is also used in the Intro to MTA guide. I added the Intro's master.adoc to the list of assemblies that include thi module.

RichardHoch commented:
@apinnick: please review.

apinnick commented on docs/topics/installing-web-console-on-openshift.adoc:
```diff
@@ -41,7 +41,7 @@ endif::[]
 .Procedure
 
-. Download the installation archive file from the link:https://developers.redhat.com/products/mta/download[{ProductShortName} download page].
+. Navigate to the  link:{MTADownloadPageURL}[{ProductShortName} Download page] and download {ProductShortName} installation file.
```
Remove extra space before link.
There should be a definite article before "... file". Also, it's a good practice to use the same names or as close as possible as the file names on the web.

Change "download {ProductShortName} installation file" to "download the Web Console `Local install & OpenShift` file." Use the appropriate attribute for Web Console.

apinnick commented on docs/topics/installing-web-console.adoc:
```diff
@@ -120,7 +120,7 @@ $ mvn clean install
 endif::[]
 ifdef::plugin-guide-offline[]
-. Download the link:{ProductDownloadURL}{IDEPluginFilename}-{ProductVersion}.zip[IDE Plugin archive file].
+. Navigate to the link:{MTADownloadPageURL}[Migration Toolkit for Applications download site] and download the IDE Plugin Repository file.
```
File name should be marked with backticks.

apinnick commented on docs/topics/maven-arguments.adoc:
```diff
@@ -134,7 +134,7 @@ The `targetTechnologies` argument now provides version support, which follows th
 When migrating to JBoss EAP, be sure to specify the version in the target, for example, `eap:6`. Specifying only `eap` will run rulesets for all versions of JBoss EAP, including those not relevant to your migration path.
 
-See link:{ProductDocGettingStartedGuideURL}#migration_paths[Supported migration paths] in the {ProductShortName} _{GettingStartedBookName}_ for which JBoss EAP version is appropriate for your source platform.
+See link:{ProductDocIntroToMTAGuideURL}#migration_paths[Supported migration paths] in the {ProductShortName} _{IntroToMTABookName}_ for which JBoss EAP version is appropriate for your source platform.
```
Fluff. I saw this before but forgot to comment on it.

Change "for which JBoss EAP version is appropriate for your source platform" to "for the appropriate JBoss EAP version." I think this appears in other places.

apinnick commented on docs/topics/rules-guide-intro.adoc:
```diff
@@ -5,6 +5,6 @@
 This guide is for engineers, consultants, and others who want to create custom XML-based rules for {ProductName} ({ProductShortName}) tools.
 
-If you are new to {ProductShortName}, it is recommended that you start with the link:{ProductDocUserGuideURL}[_{GettingStartedBookName}_] for an overview of the {ProductName} features and system requirements. It is also recommended that you review the link:{ProductDocUserGuideURL}[_{UserCLIBookName}_], which provides detailed instructions on how to install and execute the {CLIName}.
+If you are new to {ProductShortName}, read the link:{ProductDocIntroToMTAGuideURL}[_{IntroToMTABookName}_] for an overview of the {ProductName} features and system requirements and the link:{ProductDocUserGuideURL}[_{UserCLIBookName}_], which provides detailed instructions on how to install and execute the {CLIName}.
```
Fluff. Suggested change

Change "If you are new to {ProductShortName}, read the link:{ProductDocIntroToMTAGuideURL}[_{IntroToMTABookName}_] for an overview of the {ProductName} features and system requirements and the link:{ProductDocUserGuideURL}[_{UserCLIBookName}_], which provides detailed instructions on how to install and execute the {CLIName}." 

to 

"For more information, see the link:{ProductDocIntroToMTAGuideURL}[_{IntroToMTABookName}_] for an overview and the link:{ProductDocUserGuideURL}[_{UserCLIBookName}_] for details."

apinnick commented on docs/getting-started-guide/master.adoc:
```diff
@@ -8,15 +8,15 @@ include::topics/templates/document-attributes.adoc[]
 :getting-started-guide:
 
-= {GettingStartedBookName}
+= {IntroToMTABookName}
```
Can you please upload a preview to the intro guide? I don't see it listed in the links.

apinnick commented on docs/topics/installing-web-console.adoc:
```diff
@@ -63,7 +63,7 @@ endif::[]
 
 ifdef::cli-guide,web-console-guide[]
-. Download the installation archive file from the link:https://developers.redhat.com/products/mta/download[{ProductShortName} Download page].
+. Navigate to the  link:{MTADownloadPageURL}[{ProductShortName} Download page] and download the {ProductShortName} installation file.
```
Same comments as above.

RichardHoch commented:
@apinnick : changes made. Please review.

apinnick commented on docs/topics/installing-web-console.adoc:
```diff
@@ -62,8 +62,15 @@ endif::[]
 .Procedure
 
+ifdef::cli-guide[]
+. Navigate to the link:{MTADownloadPageURL}[{ProductShortName} Download page] and download the Migration Toolkit CLI `Command Line Interface` file.
```
If the name of the file is Command Line Interface, you don't need "CLI".
